### PR TITLE
add `Regex` documentation

### PR DESCRIPTION
**Pull request by ChristianKurz**
add `Regex` documentation. (#26919)

### CODE REVIEW

File: base/regex.jl [hunk 1]
```diff
@@ -7,6 +7,16 @@ include("pcre.jl")
 const DEFAULT_COMPILER_OPTS = PCRE.UTF | PCRE.NO_UTF_CHECK | PCRE.ALT_BSUX | PCRE.UCP
 const DEFAULT_MATCH_OPTS = PCRE.NO_UTF_CHECK
 
+"""
+    Regex(pattern[, flags])
+
+A type representing a regular expression. `Regex`-objects can be used to match strings
```
Comment (waldyrious) [inline base/regex.jl]: Why the hyphen? I don't think it's warranted.

File: base/regex.jl [hunk 2]
```diff
@@ -7,6 +7,16 @@ include("pcre.jl")
...
+with [`match`](@ref) or [`matchall`](@ref).
+
+`Regex` objects can be created using the [`@r_str`](@ref) string-macro. The
```
Comment (waldyrious) [inline base/regex.jl]: Here too the hyphen feels odd to me. I'd simply write "string macro".

File: base/regex.jl (hunk 1 above)
Comment (ChristianKurz) [reply base/regex.jl]: thanks for the comment!

File: base/regex.jl [hunk 3]
```diff
@@ -7,6 +7,16 @@ include("pcre.jl")
...
 
+"""
+    Regex(pattern[, flags])
+
+A type representing a regular expression. `Regex` objects can be used to match strings
+with [`match`](@ref) or [`matchall`](@ref).
+
+`Regex` objects can be created using the [`@r_str`](@ref) string macro. The
+`Regex(pattern[, flags])` constructor is usually used if the `pattern` string needs
+to be interpolated. See the documentation of the string-macro for details on flags.
```
Comment (waldyrious) [inline base/regex.jl]: You missed the hyphen here :)

File: base/regex.jl [hunk 4]
```diff
@@ -7,6 +7,16 @@ include("pcre.jl")
 const DEFAULT_COMPILER_OPTS = PCRE.UTF | PCRE.NO_UTF_CHECK | PCRE.ALT_BSUX | PCRE.UCP
 const DEFAULT_MATCH_OPTS = PCRE.NO_UTF_CHECK
 
+"""
+    Regex(pattern[, flags])
+
+A type representing a regular expression. `Regex` objects can be used to match strings
+with [`match`](@ref) or [`matchall`](@ref).
```
Comment (fredrikekre) [inline base/regex.jl]: `matchall` does not exist anymore.

File: base/regex.jl [hunk 5]
```diff
@@ -81,6 +91,8 @@ listed after the ending quote, to change its behaviour:
   `\\s`, `\\W`, `\\w`, etc. match based on Unicode character properties. With this option,
   these sequences only match ASCII characters.
 
+See `Regex` if interpolation is needed.
```
Comment (KristofferC) [inline base/regex.jl]:
```
[`Regex`](@ref)
```

maybe?

Comment (KristofferC) [reply base/regex.jl]: pushed the update

**Comment by ChristianKurz** [comment]
I had that referenc in an earlier commit, but it seems like the ref needs to be added to `doc/src/base/strings.md` and i didn't know if that file is generated or if it should be edited manually by me.

**Pull request merged**